### PR TITLE
sonic configuration management support

### PR DESCRIPTION
**Pull request by bhaveshdell**
-Add sonic-config-mgmt.yang 
-CLI xml
-mgmt-framework back end support.

### CODE REVIEW

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  revision 2019-09-15 {
+    description
+      "Initial revision.";
+  }
+
+
+  rpc copy {
+      input {
+          leaf source {
```
Comment (joyas-joseph) [inline models/yang/sonic/sonic-config-mgmt.yang:34]: Please do add a description to source and destination attributes.

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 2]
```diff
@@ -0,0 +1,39 @@
+module sonic-config-mgmt {
+  namespace "http://github.com/Azure/sonic-config-mgmt";
+  prefix scm;
+
+  organization
+    "BRCM";
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-config-mgmt.yang]: Update to "SONiC"

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  contact
+    "BRCM";
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-config-mgmt.yang]: Update to "SONiC"

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  description
+    "SONIC ERROR";
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-config-mgmt.yang]: ERROR ?

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          leaf source {
+              type string;
+          }
+
+          leaf overwrite {
+              type boolean;
+          }
+
+          leaf destination {
+              type string;
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-config-mgmt.yang]: Any pattern here ?

File: models/yang/sonic/sonic-config-mgmt.yang [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  revision 2019-09-15 {
+    description
+      "Initial revision.";
+  }
+
+
+  rpc copy {
+      input {
+          leaf source {
+              type string;
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-config-mgmt.yang]: Any pattern here ?

**Review by dutta-partha — CHANGES_REQUESTED** [review]

File: models/yang/sonic/sonic-config-mgmt.yang (hunk 3 above)
Comment (bhaveshdell) [reply models/yang/sonic/sonic-config-mgmt.yang]: Done

File: models/yang/sonic/sonic-config-mgmt.yang (hunk 4 above)
Comment (bhaveshdell) [reply models/yang/sonic/sonic-config-mgmt.yang]: Done. Fixed the text to Configuration management.

File: models/yang/sonic/sonic-config-mgmt.yang (hunk 1 above)
Comment (bhaveshdell) [reply models/yang/sonic/sonic-config-mgmt.yang:34]: Done.

File: models/yang/sonic/sonic-config-mgmt.yang (hunk 6 above)
Comment (bhaveshdell) [reply models/yang/sonic/sonic-config-mgmt.yang]: Added pattern.

File: models/yang/sonic/sonic-config-mgmt.yang (hunk 5 above)
Comment (bhaveshdell) [reply models/yang/sonic/sonic-config-mgmt.yang]: Done. Updated with pattern

File: src/CLI/clitree/cli-xml/config_mgmt.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        <COMMAND
+            name="copy"
+            help="Perform file copy operations"
+            ptype="SUBCOMMAND">
+                <PARAM name="src-copy-options"
+                       help="Copy source options"
+                       ptype="SUBCOMMAND"
+                       mode="switch">
+                       <PARAM name="copy_config_url"
+                              help="Copy directory(file://filepath)"
```
Comment (joyas-joseph) [inline src/CLI/clitree/cli-xml/config_mgmt.xml]: What does directory indicate in the help string?

**Review by joyas-joseph — APPROVED** [review]

**Review by joyas-joseph — APPROVED** [review]

**Comment by jeffyin-dell** [comment]
@dutta-partha  please provide review/approval

**Review by dutta-partha — APPROVED** [review]

**Comment by jeffyin-dell** [comment]
@joyas-joseph feel free to merge when ready

**Pull request merged**